core/app/models/spree/line_item.rb

```diff
@@ -175,7 +170,9 @@ def update_tax_charge
     end
 
     def ensure_proper_currency
-      unless currency == order.currency
+      if currency != order.currency
+        Spree::Deprecation.warn "You tried setting the currency for this line item manually." \
```

Comment:
Maybe not everyone intended to change the line item currency and something else change it. Without knowing that, the current warning could be misleading. 

Why not just: "The line items currency is different from it's order currency. This behavior is not supported anymore and will be deleted soon."?